packages/webapp/src/containers/Task/TaskAssignment/index.jsx

```diff
@@ -19,58 +26,95 @@ export default function TaskManagement({ history, match, location }) {
-        let obj = {
-          label: userData[i].first_name + ' ' + userData[i].last_name,
-          value: userData[i].user_id,
-        };
-        innerOptions.push(obj);
-        let wageObj = {
-          [userData[i].user_id]: {
-            currency: userData[i].units.currency,
-            hourly_wage: userData[i].wage.amount,
-            currencySymbol: currencySymbol,
-          },
+  const defaultAssignee = useMemo(() => {
+    let { assignee } = persistedFormData;
+    if (!assignee) {
+      if (isFarmWorker || userData.length === 1) {
+        // "current user" if he/she is a farm worker or if he/she is the only user at the farm
+        assignee = {
+          label: worker.first_name + ' ' + worker.last_name,
+          value: worker.user_id,
         };
-        wage_data.push(wageObj);
+      } else {
+        // “Unassigned” if more than one user exists
+        assignee = { label: t('TASK.UNASSIGNED'), value: null };
       }
     }
-    setOptions(options.concat(innerOptions));
-    setWageData(wageData.concat(wage_data));
-  }, []);
+    return assignee;
+  }, [persistedFormData, isFarmWorker, users, worker]);
+
+  const taskAssignForm = useTaskAssignForm({
+    user: userFarm,
+    users: userData,
+    defaultAssignee,
+    additionalFields: {
+      [OVERRIDE_HOURLY_WAGE]: persistedFormData[OVERRIDE_HOURLY_WAGE] || false,
+      [WAGE_OVERRIDE]: persistedFormData[WAGE_OVERRIDE] || null,
+      ...cloneObject(persistedFormData),
+    },
+  });
+
+  const {
+    watch,
+    selectedWorker,
+    showHourlyWageInputs,
+    setValue,
+    clearErrors,
+    currency,
+    userFarmWage,
+  } = taskAssignForm;
+
+  const currencySymbol = grabCurrencySymbol(currency);
+  const override = watch(OVERRIDE_HOURLY_WAGE);
+
+  useEffect(() => {
+    // when assignee is changed, show user farm wage as default for "wage override" input
+    if (selectedWorker.label !== t('TASK.UNASSIGNED') && userFarmWage) {
+      setValue(WAGE_OVERRIDE, userFarmWage);
+      clearErrors(WAGE_OVERRIDE);
+    }
+  }, [selectedWorker, userFarmWage]);
 
   const onSubmit = (data) => {
+    const { hourly_wage_action, assignee, hourly_wage, override_hourly_wage, wage_at_moment } =
+      data;
+    const override =
+      (!showHourlyWageInputs && override_hourly_wage) || // user has a wage but wants to override
+      (showHourlyWageInputs && hourly_wage_action === hourlyWageActions.FOR_THIS_TASK); // no user wage and set wage for this task
+
     const postData = {
       ...persistedFormData,
-      ...data,
+      ...{
+        assignee_user_id: assignee,
+        override_hourly_wage: override,
+        wage_at_moment: override
+          ? +(showHourlyWageInputs ? +hourly_wage.toFixed(2) : wage_at_moment)
+          : null,
+      },
       returnPath: location.state ? location.state.pathname : null,
     };
+    // delete data(HOURLY_WAGE_ACTION, SELECT_ALL etc) that should not be included in API request
+    Object.keys((key) => {
```

Comment:
@SayakaOno 
Maybe I don't understand the syntax. What is the object being used here?